### PR TITLE
Add support for Brotli encoding

### PR DESCRIPTION
**Pull request by sindresorhus**
When using Node.js 11.7.0 or later.

// @kevva @silverwind

### CODE REVIEW

File: source/normalize-arguments.js [hunk 1]
```diff
@@ -5,6 +5,7 @@ const is = require('@sindresorhus/is');
 const lowercaseKeys = require('lowercase-keys');
 const urlToOptions = require('./utils/url-to-options').default;
 const validateSearchParams = require('./utils/validate-search-params');
+const supportsBrotli = require('./utils/supports-brotli');
```
Comment (szmarczak) [inline source/normalize-arguments.js]: no `.default` here?

File: source/utils/supports-brotli.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+import zlib from 'zlib';
+
+export default (): boolean => typeof (zlib as any).createBrotliDecompress === 'function';
```
Comment (szmarczak) [inline source/utils/supports-brotli.ts]: Why is it a function?

File: source/normalize-arguments.js [hunk 3]
```diff
@@ -193,7 +194,7 @@ const normalize = (url, options, defaults) => {
 	}
 
 	if (options.decompress && is.undefined(headers['accept-encoding'])) {
-		headers['accept-encoding'] = 'gzip, deflate';
+		headers['accept-encoding'] = supportsBrotli ? 'gzip, deflate, br' : 'gzip, deflate';
```
Comment (szmarczak) [inline source/normalize-arguments.js]: If `supportsBrotli` needs to be a function then it should be `supportsBrotli()`

File: source/utils/supports-brotli.ts (hunk 2 above)
Comment (sindresorhus) [reply source/utils/supports-brotli.ts]: Mistake

File: source/normalize-arguments.js (hunk 1 above)
Comment (sindresorhus) [reply source/normalize-arguments.js]: Yes, mistake

**Pull request merged**